lib/src/rules/prefer_const_constructors_in_immutables.dart

```diff
@@ -120,20 +120,22 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   bool _hasImmutableAnnotation(ClassElement clazz) {
     var selfAndInheritedClasses = _getSelfAndInheritedClasses(clazz);
-    var selfAndInheritedAnnotations =
-        selfAndInheritedClasses.expand((c) => c.metadata).map((m) => m.element);
-    return selfAndInheritedAnnotations.any(_isImmutable);
+    for (var cls in selfAndInheritedClasses) {
+      if (cls.metadata.any((m) => _isImmutable(m.element))) return true;
+    }
+    return false;
   }
 
   bool _hasMixin(ClassElement clazz) => clazz.mixins.isNotEmpty;
 
-  static Iterable<ClassElement> _getSelfAndInheritedClasses(
-      ClassElement self) sync* {
+  static Iterable<ClassElement> _getSelfAndInheritedClasses(ClassElement self) {
+    var elements = <ClassElement>[];
```

Comment:
You don't actually need `elements` because `seenElements` is `LinkedHashSet`, so it preserves the order (if you need it at all), and you could `return seenElements.toList()` instead.

Reply:
Ah!  Great catch.  Thanks.